addons/test_mail/tests/test_mail_composer.py

```diff
@@ -2028,12 +2028,12 @@ def test_mail_composer_wtpl_recipients_email_fields(self):
         #   there are more partners than email to notify;
         self.assertEqual(len(self._new_mails), 2, 'Should have created 2 mail.mail')
         self.assertEqual(
-            len(self._mails), (len(new_partners) + 2) * 2,
-            f'Should have sent {(len(new_partners) + 2) * 2} emails, one / recipient ({len(new_partners)} mailed partners + partner_1 + partner_2) * 2 records')
-        for record in self.test_records:
+            len(self._mails), len(new_partners) + 2,
+            f'Should have sent {len(new_partners) + 2} emails, one / recipient ({len(new_partners)} mailed partners + partner_1 + partner_2)')
+        for record, status in zip(self.test_records, ('sent', 'cancel')):
```

Comment:
Why is the second one canceled ? Is the content exactly the same ? In that case would probably update template content, as purpose here is to really test a multi send, not really duplicate check, would prefer keeping "sent" emails.

Reply:
It is sending the same template to both test_records, the second one is cancelled for mail_dup. I've tried for 2 hours to find a solution but I achieved nothing. Do you have a suggestion?

Reply:
In 16, duplication is done only on email (not on actual content like in newer versions, where we check attachments, subject, ...). You can use the ``mailing_document_based`` context key to remove the duplicate check in this case (as we are not trying to check duplicate check here, more email parsing). It was added in stable to indicate there are documents where same email does not mean duplicate (e.g. invoice, sale orders, ...).